src/oscar/views/generic.py

```diff
@@ -27,7 +27,15 @@ def post(self, request, *args, **kwargs):
                 return self.response
             else:
                 messages.error(request, _("Invalid form submission"))
-        return super().post(request, *args, **kwargs)
+                return self.get()
```

Comment:
Need to pass `request, *args, **kwargs` here, otherwise this will fail :-). Probably worth adding a test to make sure this works.

Reply:
There are no existing tests for any of this, so here isnt a good spot for me to break my teeth on helping write good *unit* test coverage.

I'd much rather be spending the time on https://github.com/django-oscar/django-oscar/issues/3420 , which would provide adequate test coverage for this specific case, and all of the other errors stemming from throwing ill-formed requests at oscar.

Reply:
OK - happy to merge this now. It's an edge case anyway, unlikely to affect a real user.